deeppavlov/models/multitask_pal_bert/modeling.py

```diff
@@ -0,0 +1,700 @@
+                (e.g., 512 or 1024 or 2048).
+            type_vocab_size: The vocabulary size of the `token_type_ids` passed into
+                `BertModel`.
+            initializer_range: The sttdev of the truncated_normal_initializer for
+                initializing all weight matrices.
+        """
+        self.vocab_size = vocab_size
+        self.hidden_size = hidden_size
+        self.num_hidden_layers = num_hidden_layers
+        self.num_attention_heads = num_attention_heads
+        self.hidden_act = hidden_act
+        self.intermediate_size = intermediate_size
+        self.hidden_dropout_prob = hidden_dropout_prob
+        self.attention_probs_dropout_prob = attention_probs_dropout_prob
+        self.max_position_embeddings = max_position_embeddings
+        self.type_vocab_size = type_vocab_size
+        self.initializer_range = initializer_range
+        self.hidden_size_aug = hidden_size_aug
+        self.pals = pals
+        self.extra_dim = extra_dim
+        self.houlsby = houlsby
+        self.mult = mult
+        self.top = top
+        self.bert_lay_top = bert_lay_top
+        self.lhuc = lhuc
+        self.num_tasks = num_tasks
+
+    @classmethod
+    def from_dict(cls, json_object):
+        """Constructs a `BertConfig` from a Python dictionary of parameters."""
+        config = BertConfig(vocab_size=None)
+        for (key, value) in six.iteritems(json_object):
+            config.__dict__[key] = value
+        return config
+
+    @classmethod
+    def from_json_file(cls, json_file):
+        """Constructs a `BertConfig` from a json file of parameters."""
+        with open(json_file, "r") as reader:
+            text = reader.read()
+        return cls.from_dict(json.loads(text))
+
+    def to_dict(self):
+        """Serializes this instance to a Python dictionary."""
+        output = copy.deepcopy(self.__dict__)
+        return output
+
+    def to_json_string(self):
+        """Serializes this instance to a JSON string."""
+        return json.dumps(self.to_dict(), indent=2, sort_keys=True) + "\n"
+
+
+class BERTLayerNorm(nn.Module):
+    def __init__(self, config, multi_params=None, variance_epsilon=1e-12):
+        """Construct a layernorm module in the TF style (epsilon inside the square root)."""
+        super(BERTLayerNorm, self).__init__()
+        if multi_params is not None:
+            self.gamma = nn.Parameter(torch.ones(config.hidden_size_aug))
+            self.beta = nn.Parameter(torch.zeros(config.hidden_size_aug))
+        else:
+            self.gamma = nn.Parameter(torch.ones(config.hidden_size))
+            self.beta = nn.Parameter(torch.zeros(config.hidden_size))
+        self.variance_epsilon = variance_epsilon
+
+    def forward(self, x):
+        u = x.mean(-1, keepdim=True)
+        s = (x - u).pow(2).mean(-1, keepdim=True)
+        x = (x - u) / torch.sqrt(s + self.variance_epsilon)
+        return self.gamma * x + self.beta
+
+
+class BERTEmbeddings(nn.Module):
```

Comment:
I think it's better to remove this class to torch_bert ( e.g in modeling.py file) as it's not directly related to the multitasking

Reply:
> I think it's better to remove this class to torch_bert ( e.g in modeling.py file) with everything that can be used without multitasking

I think this is the only class that does have anything to do with multitask specific parameters. I will create a new file modeling.py and move this class there.

Reply:
Actually, this class does use `BERTLayerNorm` class which does use `multi_params`, so if I move this class I will have to import it in torch_bert as well. So I think keeping it here would be better, what do you think?

Reply:
I think that everything that can be used without multitasking ( including BertLayerNorm that can be used as usual with multi_params None) is better to be placed there

Reply:
Moved unrelated classes to a different location.